harness/determined/exec/tensorboard.py

```diff
@@ -26,121 +38,140 @@ def set_s3_region() -> None:
+def get_tensorboard_args(tb_version: str, tfevents_dir: str, add_args: List[str]) -> List[str]:
+    """Build tensorboard startup args.
 
-
-def get_tensorboard_version(version: str) -> Tuple[str, str]:
-    """
-    Gets the version of the tensorboard package currently installed. Used
-    by downstream processes to determine args passed in.
-    :return: version in the form of (major, minor) tuple
-    """
-
-    major, minor, _ = version.split(".")
-
-    return major, minor
-
-
-def get_tensorboard_args(args: List[str]) -> List[str]:
-    """
-    Builds tensorboard startup args from args passed in from tensorboard-entrypoint.sh
     Args are added and deprecated at the mercy of tensorboard; all of the below are necessary to
     support versions 1.14, 2.4, and 2.5
 
-    - If multiple directories are specified and the tensorboard version is > 1,
-    use legacy logdir_spec behavior
-
     - Tensorboard 2+ no longer exposes all ports. Must pass in "--bind_all" to expose localhost
-
     - Tensorboard 2.5.0 introduces an experimental feature (default load_fast=true)
     which prevents multiple plugins from loading correctly.
     """
     task_id = os.environ["DET_TASK_ID"]
     port = os.environ["TENSORBOARD_PORT"]
 
-    version = args.pop(0)
-
-    # logdir is the second argument passed in from tensorboard_manager.go. If multiple directories
-    # are specified and the tensorboard version is > 1, use legacy logdir_spec behavior. NOTE:
-    # legacy logdir_spec behavior is not supported by many tensorboard plugins
-    logdir = args.pop(0)
-
     tensorboard_args = [
         "tensorboard",
         f"--port={port}",
         f"--path_prefix=/proxy/{task_id}",
-        *args,
+        *add_args,
     ]
 
-    major, minor = get_tensorboard_version(version)
+    # Version dependant args
+    version_parts = tb_version.split(".")
+    major = int(version_parts[0])
+    minor = int(version_parts[1])
 
-    if major == "2":
+    if major >= 2:
         tensorboard_args.append("--bind_all")
-        if minor >= "5":
-            tensorboard_args.append("--load_fast=false")
-        if len(logdir.split(",")) > 1:
-            tensorboard_args.append(f"--logdir_spec={logdir}")
-            return tensorboard_args
+    if major > 2 or major == 2 and minor >= 5:
+        tensorboard_args.append("--load_fast=false")
 
-    tensorboard_args.append(f"--logdir={logdir}")
-    return tensorboard_args
+    tensorboard_args.append(f"--logdir={tfevents_dir}")
 
+    return tensorboard_args
 
-def main(args: List[str]) -> int:
-    set_s3_region()
 
+def get_tensorboard_url() -> str:
     task_id = os.environ["DET_TASK_ID"]
     port = os.environ["TENSORBOARD_PORT"]
     tensorboard_addr = f"http://localhost:{port}/proxy/{task_id}"
-    url = f"{tensorboard_addr}/data/plugin/scalars/tags"
-    tensorboard_args = get_tensorboard_args(args)
+    return f"{tensorboard_addr}/data/plugin/scalars/tags"
+
+
+def check_for_metrics() -> bool:
```

Comment:
Why wait for tensorboard to show metrics, when we now know perfectly well when metrics are ready?

Reply:
I am not sure what you mean here, could you add a little clarity?

You could me that we know if we have downloaded files or not, so why wait to see if Tensorboard is responding with metrics?

The short answer is, this was how it was working.

Reply:
> You could me that we know if we have downloaded files or not, so why wait to see if Tensorboard is responding with metrics?

Yes.  I know how it was working but there's a better way now, right?

Reply:
What would be the better way? Just launch it? Say the word and I will. It solves another issue which is that I think it hangs when someone is trying to use the profiler.

Reply:
So, as soon as one download happens, I will trigger the ready message.